### ISSUE TITLE
Resource Sensor not declaring state correctly.

### ISSUE THREAD

**Issue by Caseyckk** [Berkays/OniMods#2]
Testing it a bit more it seems there are two issues. 
If there is 0 of the items in the room/area then it will always be red regardless of settings.
![image](https://user-images.githubusercontent.com/33268058/210574618-5d4ce5d6-bc4b-4a25-b3e5-94b22286a90e.png)

Second is that it appears to ignore it's own readings of an item. So I put a sensor on my second planet to tell the first planet that it needs food sent. But I noticed that on global it detects food from the first planet and reports 122kg but still remains red.
![image](https://user-images.githubusercontent.com/33268058/210574266-26d23373-e5a4-4a87-b402-1a90eeecde42.png)

Both states/situations I unpaused and ran for several moments to allow them to update to any changes.

**Comment by Berkays** [comment]
Thanks for the report. I couldn't reproduce the second issue you mentioned. Can you try this version to see if it fixes it?

Replace the files in mods/Steam/2911545239/
[ResourceSensor.zip](https://github.com/Berkays/OniMods/files/10345685/ResourceSensor.zip)

**Comment by Caseyckk** [comment]
TLDR; Seems to work, slap a new version number on that build and everyone might be happy. Or less unhappy, don't let me tell them how to live.


Dropped it in, I had to fight with steam a little to not update it (ironic considering how hard it is to get steam to update mods sometimes). Anyway I think I'm using the new version but not 100% since it's same version #.

First issue seems solved, Global (ie that planet only) it is detecting the 70.6 kg of liceloaf there and sending green since that is below 402kg. I flipped it and it went red so both states appear to work as expected.
![image](https://user-images.githubusercontent.com/33268058/210623474-e5c3e0f6-9dc3-4aaa-9aaf-d90dee9ffac0.png)

Second issue, first I noticed it now complains about not being in a room, changing it to distance appears to have it work inline with its display now. I checked the screenshots and before it didn't make mention of requiring a room but perhaps that should have been obvious when using the "room" mode. I would note however that the fridge does have 70.6 kg of liceloaf so that is not being detected. I assume either because that is a different mod or in a container but if room mode is purely for detecting objects in the world than this works as expected.
![image](https://user-images.githubusercontent.com/33268058/210624696-812c221b-038e-4175-ac8e-d1d1740a72ed.png)

I built a new one in a room to test room mode and it appears to work just fine too.

**Comment by Berkays** [comment]
Yeah room mode uses the games definition of "room" so if you can't see it in the room overlay as at least miscellaneous room it won't count any materials. This is not easily solvable as having to manually calculate rooms will have performance impacts. Maybe in conjuction with another mod that increases misc room size limits it can have a workaround.

It doesn't count any resources in any storage buildings for any mode. That can be added as another option maybe. You can still count those with a smart storage bin maybe (I know it's finnicky to power all those bins but this gives them a use)
